src/GravityFormsExtensions.php

```diff
@@ -463,4 +505,65 @@ private function find_nested_blocks(string $block_name, array $blocks): array
 
         return $matching_blocks;
     }
+
+    /**
+     * Redirect using Javascript after form submission instead of sending a header.
+     * Makes it possible to send tag manager events before redirecting.
+     *
+     * @param string|array $confirmation The default confirmation message.
+     * @param mixed        $form         The form properties.
+     *
+     * @return string|array The custom confirmation message.
+     */
+    public function p4_gf_custom_confirmation_redirect($confirmation, $form)
+    {
+        GFCommon::log_debug(__METHOD__ . '(): running.');
```

Comment:
Do we want to keep these `log_debug` calls?

Reply:
Its normal to add a log_debug() in GF, the default GF debug log feature is disable.
More info about GF custom debug logs:
https://docs.gravityforms.com/custom-logging-statements/
https://docs.gravityforms.com/logging-and-debugging/